packages/blocks/tests/php/test-blocks.php

```diff
@@ -0,0 +1,69 @@
+<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
+/**
+ * Test methods from Automattic\Jetpack\Blocks
+ *
+ * @since 9.0.0
+ *
+ * @package automattic/jetpack-blocks
+ */
+
+namespace Automattic\Jetpack;
+
+use Automattic\Jetpack\Blocks;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Class Test_Blocks
+ */
+class Test_Blocks extends TestCase {
+	/**
+	 * Test the different inputs and matching output for Classes.
+	 *
+	 * @since 9.0.0
+	 *
+	 * @covers Automattic\Jetpack\Blocks::classes
+	 */
+	public function test_block_classes() {
+		$block_name = 'foo';
+		$attr       = array(
+			'bar'       => 'baz',
+			'align'     => 'wide',
+			'className' => 'editorclass',
+		);
+		$extra      = array( 'extraclass' );
+
+		$block_classes = Blocks::classes( $block_name, $attr, $extra );
+
+		$this->assertContains( 'wp-block-jetpack-foo', $block_classes ); // a general class is created from the block name.
+		$this->assertNotContains( 'bar', $block_classes ); // The extra 'foo' attribute should be dropped.
+		$this->assertContains( 'alignwide', $block_classes ); // an alignment class is created.
+		$this->assertContains( 'editorclass', $block_classes ); // className classes are passed.
+		$this->assertContains( 'extraclass', $block_classes ); // Extra class remains.
+	}
+
+	/**
+	 * Test whether we can detect an AMP view.
+	 *
+	 * @since 9.0.0
+	 *
+	 * @covers Automattic\Jetpack\Blocks::is_amp_request
+	 */
+	public function test_is_amp_request() {
+		add_filter( 'jetpack_is_amp_request', '__return_true' );
+
+		$this->assertTrue( Blocks::is_amp_request() );
+
+		remove_filter( 'jetpack_is_amp_request', '__return_true' );
```

Comment:
I note that if for some reason the test fails, this line won't be reached since PHPUnit will throw an exception to indicate the failure. If that means the filter doesn't get removed, would that screw up future tests? On the other hand, if the filter *does* get removed by the testing framework, does that mean this line is redundant?

Reply:
That's a good point. This should be addressed in 02bd21d0be285cd0af85e71c7fba09a0de52ec3c